Multisig/Logic/Models/TokenBalance.swift

```diff
@@ -35,16 +35,14 @@ extension TokenBalance {
 
     init(address: Address, symbol: String?, logoUri: String?, tokenBalance: UInt256String, decimals: UInt256String?, fiatBalance: String, code: String) {
         self.address = address.checksummed
-        self.symbol = symbol ?? "ETH"
-        self.imageURL = logoUri.flatMap { URL(string: $0) }
+        let coin = Network.nativeCoin!
 
-        if address == .ether {
-            image = UIImage(named: "ico-ether")
-        }
+        self.symbol = symbol ?? coin.symbol!
+        self.imageURL = logoUri.flatMap { URL(string: $0) } ?? coin.logoUrl
 
         let tokenFormatter = TokenFormatter()
         let amount = Int256(tokenBalance.value)
-        let precision = decimals?.value ?? 18 // ETH
+        let precision = decimals?.value ?? UInt256(coin.decimals)
```

Comment:
can't we just define the type of decimals to be UInt256 instead of casting it?

Reply:
@MouazAlzahabi it's going to be part of a different ticket